patches/server/0882-Added-EntityToggleSitEvent.patch

```diff
@@ -0,0 +1,48 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: KyGuy2002 <IEatBeans#1165>
+Date: Fri, 11 Mar 2022 15:33:10 +0000
+Subject: [PATCH] Added EntityToggleSitEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/TamableAnimal.java b/src/main/java/net/minecraft/world/entity/TamableAnimal.java
+index 7e67c47434d1f29cc87f3af926f3efc920c87829..b4a958f0a7380772f4b0d71a1aef3b8fa3fb8157 100644
+--- a/src/main/java/net/minecraft/world/entity/TamableAnimal.java
++++ b/src/main/java/net/minecraft/world/entity/TamableAnimal.java
+@@ -126,6 +126,12 @@ public abstract class TamableAnimal extends Animal implements OwnableEntity {
+     }
+ 
+     public void setInSittingPose(boolean inSittingPose) {
++        // Paper start - call EntityToggleSitEvent
++        io.papermc.paper.event.entity.EntityToggleSitEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityToggleSitEvent(this, inSittingPose);
++        if (event.isCancelled()) {
++            return;
++        }
++        // Paper end
+         byte b = this.entityData.get(DATA_FLAGS_ID);
+         if (inSittingPose) {
+             this.entityData.set(DATA_FLAGS_ID, (byte)(b | 1));
+diff --git a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+index a3c8a0291fa9ae6f3c96d937dd4621edd7c48535..dd4a5bfa6d2ddc3532d3ab174c8044f71949c88f 100644
+--- a/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
++++ b/src/main/java/org/bukkit/craftbukkit/event/CraftEventFactory.java
+@@ -12,6 +12,8 @@ import java.util.List;
+ import java.util.Map;
+ import java.util.stream.Collectors;
+ import javax.annotation.Nullable;
++
++import io.papermc.paper.event.entity.EntityToggleSitEvent;
+ import net.minecraft.core.BlockPos;
+ import net.minecraft.core.Direction;
+ import net.minecraft.network.protocol.game.ServerboundContainerClosePacket;
+@@ -1918,5 +1920,11 @@ public class CraftEventFactory {
+         io.papermc.paper.event.block.BlockPreDispenseEvent event = new io.papermc.paper.event.block.BlockPreDispenseEvent(block, org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(itemStack), slot);
+         return event.callEvent();
+     }
++
++    public static EntityToggleSitEvent callEntityToggleSitEvent(Entity entity, Boolean isSitting) {
++        io.papermc.paper.event.entity.EntityToggleSitEvent event = new io.papermc.paper.event.entity.EntityToggleSitEvent(entity.getBukkitEntity(), isSitting);
++        Bukkit.getPluginManager().callEvent(event);
++        return event;
++    }
```

Comment:
for just one location, with no special logic needed along with it, this is unneeded, just call the event right where it is.